CMakeLists.txt

```diff
@@ -972,6 +978,18 @@ else()
   set(NO_WHOLE_ARCHIVE_LIBS kernel)
 endif()
 
+# Move the libraries that are supposed to be linked late in the link
+# process to the end of their respective list.
+foreach(zephyr_lib ${ZEPHYR_LATE_LIBS_PROPERTY})
+  if("${zephyr_lib}" IN_LIST WHOLE_ARCHIVE_LIBS)
+    list(REMOVE_ITEM WHOLE_ARCHIVE_LIBS ${zephyr_lib})
+    list(APPEND WHOLE_ARCHIVE_LIBS ${zephyr_lib})
+  elseif("${zephyr_lib}" IN_LIST NO_WHOLE_ARCHIVE_LIBS)
+    list(REMOVE_ITEM NO_WHOLE_ARCHIVE_LIBS ${zephyr_lib})
+    list(APPEND NO_WHOLE_ARCHIVE_LIBS ${zephyr_lib})
+  endif()
+endforeach()
```

Comment:
let's not re-invent something already well handled by CMake.
Also this will become whac-a-mole in the future when order starts to matter and we then need to guarantee an order, ref:
> Ordering between late libs is not guaranteed.

From the PR description, then I would like to ask, why is the `llext-exports` library a Zephyr library in the first place ?
From the looks of this PR then it seems to me that the `llext-exports` lib should be outside the `--whole-archive` and thus not be a Zephyr library.

Keeping the lib which needs this LATE LINKING control outside whole-archive allows to user regular `target_link_libraries()` to ensure correct link order.